test/test_tensor_creation_ops.py

```diff
@@ -3171,11 +3171,12 @@ def test_rand(self, device, dtype):
             torch.rand(size, size, out=res2)
             self.assertEqual(res1, res2)
 
-    @onlyCUDA
     def test_randperm(self, device):
-        if device == 'cpu':
+        if device == 'cpu' or device == 'meta':
```

Comment:
This seems like an OK bandaid, but why not?

```
rng_device = None
if torch.device(device).type == 'cuda':
  rng_device = [device]
```

Reply:
Because it still won't work. You can't actually afford to skip seeding the non-CUDA RNG:

```
                with torch.random.fork_rng(devices=rng_device):
                    res1 = torch.randperm(n, dtype=dtype, device=device)
                res2 = torch.empty(0, dtype=dtype, device=device)
                torch.randperm(n, out=res2, dtype=dtype, device=device)
                self.assertEqual(res1, res2, atol=0, rtol=0)
```

Exclusion is OK for CPU (as `fork_rng` always reseeds CPU) and meta (which doesn't actually use the RNG). But anything else will still fail the test because they didn't get their RNG forked.